components/director/internal/domain/scenarioassignment/repository.go

```diff
@@ -111,6 +113,31 @@ func (r *repository) GetForScenarioName(ctx context.Context, tenantID, scenarioN
 	return assignmentModel, nil
 }
 
+// ListForScenarioNames list ASAs for given scenario names
+func (r *repository) ListForScenarioNames(ctx context.Context, tenantID string, scenarioNames []string) ([]*model.AutomaticScenarioAssignment, error) {
+	var collection EntityCollection
+
+	scenarioConditions := make(repo.Conditions, 0, len(scenarioNames))
+	for _, name := range scenarioNames {
+		scenarioConditions = append(scenarioConditions, repo.NewEqualCondition(scenarioColumn, name))
+	}
+
+	conditions := repo.Or(repo.ConditionTreesFromConditions(scenarioConditions)...)
+
+	if err := r.conditionLister.ListConditionTree(ctx, resource.AutomaticScenarioAssigment, tenantID, &collection, conditions); err != nil {
```

Comment:
You can use the standard lister with 	`repo.NewInConditionForStringValues(scenarioColumn, scenarioNames)` instead of this one